iree/compiler/Dialect/Flow/Transforms/RematerializeDispatchConstants.cpp

```diff
@@ -212,9 +255,18 @@ class RematerializeDispatchConstantsPass
       // Note: we iterate in reverse so that the rematerialized constants appear
       // in the same order they did originally (as insertion is at the top).
       for (auto constantOp : llvm::reverse(smallConstantOps)) {
```

Comment:
Just FYI because I haven't written it down anywhere: this algorithm is inverted and ideally it would look for each region op and then each arg to see if it's constant so that the complexity scales with O(dispatch regions), whereas right now it's O(constants) even if there are no dispatch regions in the function. Not sure what I was thinking when I wrote this. If you ever see compile time here that's what it is :)

Reply:
Correct! this need to be a `DispatchOp` pattern rewrite which will be O(dispatch ops). I am just getting things flowing in the tensor land this can be simple refactoring after we nuke all the old dispatch region/buffer path  =)